pkg/controllers/runtime_controller.go

```diff
@@ -223,6 +223,27 @@ func (r *RuntimeReconciler) ReconcileRuntime(engine base.Engine, ctx cruntime.Re
 		}
 	} else {
 		log.Info("The runtime is already setup.")
+		engine, ok := engine.(*base.TemplateEngine)
```

Comment:
Perhaps we can refer to the implementation of Setup so that we not need to transform base.engine to TemplateEngine